examples/jsm/renderers/webgl/nodes/WebGLNodes.js

```diff
@@ -47,3 +46,39 @@ Material.prototype.onBeforeRender = function ( renderer, scene, camera, geometry
 	}
 
 };
+
+WebGLRenderer.prototype.compute = async function ( ...computeNodes ) {
+
+	const material = new MeshBasicNodeMaterial();
+	const geometry = new PlaneGeometry( 2, 2 );
+	const object = new Mesh( geometry, material );
+	const scene = new Scene().add( object );
+	const camera = new Camera();
+
+	for ( const computeNode of computeNodes ) {
+
+		material.colorNode = computeNode.computeNode;
+		material.needsUpdate = true;
+
+		this.compile( scene, camera ); // Compile material and populate outComputeBuffer
+
+		const outBuffer = builders.get( material ).outComputeBuffer;
+
+		const renderTarget = outBuffer.renderTarget;
+		const currentRenderTarget = this.getRenderTarget();
+		this.setRenderTarget( renderTarget );
+		// nodeFrame.update();
+		this.render( scene, camera );
+		if ( computeNode.populateOutArray === true ) { // Note that if this is false then the results will be available only for further computeNodes, and will not be available after the completion of WebGLRenderer.compute()
+
+			this.readRenderTargetPixels( renderTarget, 0, 0, renderTarget.width, renderTarget.height, outBuffer.attribute.array );
```

Comment:
Ideally we could use `getArrayBuffer()` for this like WebGPURenderer.

Reply:
Unfortunately, I think, no -- such `getArrayBuffer` should work with the render target assigned to the attribute, so it needs WebGLBuffer, but it's only stored in an "unknown" material's NodeBuilder... Maybe we can add a property to WebGLRenderer that stores all WebGLBuffers -- then that will solve this problem and also will allow disposing them.

Reply:
Implemented!